resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerState.java

```diff
@@ -32,7 +32,9 @@
         this.stateMachine = stateMachine;
     }
 
-    abstract boolean isCallPermitted();
+    abstract boolean obtainPermission();
```

Comment:
Having done some C# programming I would actually switch how you implemented them, `tryObtainPermission()` should return `boolean` and `obtainPermission()` should throw an exception.

See https://stackoverflow.com/questions/467613/parse-v-tryparse

Reply:
Yes makes sense.

Reply:
Be aware that the PR is still work in progress. But feel free to review  it